csharp/src/Apache.Arrow/Types/ListType.cs

```diff
@@ -25,10 +25,16 @@ public sealed class ListType : ArrowType
         public Field ValueField { get; }
         public IArrowType ValueDataType { get; }
 
-        public ListType(Field valueField, IArrowType valueDataType)
+        public ListType(Field valueField)
         {
             ValueField = valueField ?? throw new ArgumentNullException(nameof(valueField));
+            ValueDataType = ValueField.DataType;
+        }
+
+        public ListType(IArrowType valueDataType)
```

Comment:
Can this constructor just call into the constructor above? Similar to the way it works in C++:

https://github.com/apache/arrow/blob/81909dc0775a686cb8d89037dc96d2cd78e01a1e/cpp/src/arrow/type.h#L591-L597

Reply:
Thanks for your suggestion.
I modified it.